src/core/internal/convert.d

```diff
@@ -633,3 +633,41 @@ const(ubyte)[] toUbyte(T)(ref T val) if (is(T == struct) || is(T == union))
         return (cast(const(ubyte)*)&val)[0 .. T.sizeof];
     }
 }
+
+char[] itoa(ulong n) pure nothrow @safe
```

Comment:
This seems to be a duplicate of something in demangle.d:
https://github.com/D-Programming-Language/druntime/blob/5afecf8d7aa21f3269298767ec8d2c753103e6ba/src/core/demangle.d#L1706

It's implementation seems to be more efficient: It looks like you are writting the digits lsd to msd, and then reversing the digits? Instead, you can write the digits msd to lsd, and dup `[20 - length .. length]`.

Bonus points for you though because you seem to handle the case where n == 0, whereas the demangle one does not seem to?

I get that returning a `char[]` means purity can make it a string, but it also means that you will always allocate for things such as "0". I guess it is moot here though.

Both functions are used differently, but would it be possible to consolidate the implementations somewhat?